crates/wasi-provider/src/wasi_runtime.rs

```diff
@@ -137,14 +256,25 @@ mod test {
             HashMap::default(),
             Vec::default(),
             HashMap::default(),
+            std::env::current_dir().unwrap(),
         )
         .expect("wasi runtime init");
-        let output = NamedTempFile::new().unwrap();
-        wr.run(output.reopen().unwrap()).expect("complete run");
-        let mut stdout = String::default();
+        let mut handle = wr.start().await.expect("runtime handle");
+        handle.wait().await.expect("successful run");
+
+        let mut output = Vec::new();
+        handle.output(&mut output).await.unwrap();
+        assert_eq!("Hello, world!\n".to_string().into_bytes(), output);
+
+        let status = handle.status().await.unwrap();
+        assert!(match status {
```

Comment:
Can't wait for CI to update to the latest Rust so we can use `matches!`

Reply:
I had that in there but their next release is 4-5 days out